### DISCUSSION TITLE
docs state different `matchCurrentVersion` values for same context

### DISCUSSION THREAD

**Discussion by jmartens** [renovatebot/renovate#32735] [Request Help]
### How are you running Renovate?

None

### Please tell us more about your question or problem

It seems at autoMerge docs state the [`matchCurrentVersion` setting for auto merging non-major updates](https://docs.renovatebot.com/key-concepts/automerge/#automerge-non-major-updates)  is defined as:

```
"matchCurrentVersion": "!/^0/"
```

where at the [`:automergeNonMajor` preset]() it seems to be defined as:

```
"matchCurrentVersion": ">= 1.0.0",
``` 

Technically they do the same of course, but IMHO this is inconsequent.

### Logs (if relevant)

n/a

**Comment by rarkins** [comment]
I think the first one is better. The second one was probably written first, and might not work for all versioning schemes

**Reply by JuJup** [reply]
I just stumbled upon the fact, that the `:automergeStableNonMajor` [preset](https://docs.renovatebot.com/presets-default/#automergestablenonmajor) is not behaving as expected, as version ranges are not supported in most of my dependencies.
Maybe the Preset could be updated and also we could add a warning box in the Configuration Docs for [matchCurrentVersion](https://docs.renovatebot.com/configuration-options/#matchcurrentversion), saying "When using a range matcher, please make sure, your versioning supports ranges" or something else?

**Reply by rarkins** [reply]
The reason the regex one is better is that it doesn't rely on versioning - it does a string regex match. You're welcome to submit a PR updating it

**Reply by JuJup** [reply]
--> https://github.com/renovatebot/renovate/pull/32924
